internal/pkg/service/templates/api/service/service.go

```diff
@@ -609,8 +609,9 @@ func repositoryRef(d dependencies.ProjectRequestScope, name string) (model.Templ
 		return repo, nil
 	} else {
 		return model.TemplateRepository{}, &GenericError{
-			Name:    "templates.repositoryNotFound",
-			Message: fmt.Sprintf(`Repository "%s" not found.`, name),
+			StatusCode: http.StatusNotFound,
```

Comment:
So you're changing status codes in some cases. Are these status codes correctly documented in `design.go` for the relevant endpoints?

Reply:
I see only `RepositoryIndexTemplatesRepositoryNotFoundResponseBody` but that's body for Repository not found for `RepositoryIndex` api endpoint. I see that these functions return `GenericError`. I see in docs that the endpoint can return `404` but we returned `500`. Maybe this is just mistake in our codebase?

Reply:
Probably, yeah.